[Bug]: Provider configuration through environment variables broken since 1.6.0

### Terraform Version

OpenTofu v1.11.5
on linux_amd64

### Terraform Configuration

```
env | grep -i outscale_
OUTSCALE_ACCESSKEYID=REDACTED
OUTSCALE_SECRETKEYID=REDACTED
OUTSCALE_REGION=cloudgouv-eu-west-1
```
```
terraform {
required_providers {
    outscale = {
      source  = "outscale/outscale"
      version = "1.6.0"
    }
}

provider "outscale" {}
```


### Debug Output (optional)

1.5.0 → 200 OK
```
2026-06-03T17:11:30.547+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0: 2026/06/03 17:11:30 [DEBUG] Waiting for state to become: [success]
2026-06-03T17:11:30.547+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0: 2026/06/03 17:11:30
2026-06-03T17:11:30.547+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0: POST /api/v1/ReadPublicIps HTTP/1.1
2026-06-03T17:11:30.547+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0: Host: api.cloudgouv-eu-west-1.outscale.com
2026-06-03T17:11:30.547+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0: User-Agent: terraform-provider-outscale/dev
2026-06-03T17:11:30.547+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0: Content-Length: 43
2026-06-03T17:11:30.547+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0: Accept: application/json
2026-06-03T17:11:30.547+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0: Content-Type: application/json
2026-06-03T17:11:30.547+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0: Accept-Encoding: gzip
2026-06-03T17:11:30.547+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0
2026-06-03T17:11:30.547+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0: {"Filters":{"PublicIps":["REDACTED"]}}
2026-06-03T17:11:30.547+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0
2026-06-03T17:11:30.942+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0: 2026/06/03 17:11:30
2026-06-03T17:11:30.942+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0: HTTP/1.1 200 OK
2026-06-03T17:11:30.942+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0: Content-Length: 341
2026-06-03T17:11:30.942+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0: Access-Control-Allow-Origin: *
2026-06-03T17:11:30.942+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0: Connection: keep-alive
2026-06-03T17:11:30.942+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0: Content-Type: application/json
2026-06-03T17:11:30.942+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0: Date: Wed, 03 Jun 2026 15:11:30 GMT
2026-06-03T17:11:30.942+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0: Referrer-Policy: same-origin
2026-06-03T17:11:30.942+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0: Strict-Transport-Security: max-age=31536000; includeSubdomains;
2026-06-03T17:11:30.942+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0: X-Content-Type-Options: nosniff
2026-06-03T17:11:30.942+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0: X-Frame-Options: SAMEORIGIN
2026-06-03T17:11:30.942+0200 [DEBUG] provider.terraform-provider-outscale_v1.5.0: X-Xss-Protection: 1; mode=block
```

1.6.0 → Note the 401 and the OUTSCALE_REGION env var that's not working either (calls to api.eu-west-2)
```
2026-06-03T17:27:31.645+0200 [DEBUG] provider.terraform-provider-outscale_v1.6.0: SDK HTTP request: tf_req_id=c99886ec-6b9c-8ec3-ac9c-4d2c614b0d25 tf_rpc=ReadDataSource @caller=github.com/outscale/terraform-provider-outscale/internal/logging/tflog_wrapper.go:44
  req=
  | POST https://api.eu-west-2.outscale.com/api/v1/ReadPublicIps
  |
  | {
  |   "Filters": {
  |     "PublicIps": [
  |       "REDACTED"
  |     ]
  |   }
  | }
   tf_data_source_type=outscale_public_ip tf_provider_addr=registry.terraform.io/outscale/outscale @module=outscale tf_mux_provider=tf5to6server.v5tov6Server timestamp="2026-06-03T17:27:31.645+0200"
2026-06-03T17:27:31.891+0200 [ERROR] provider.terraform-provider-outscale_v1.6.0: SDK HTTP response error: @module=outscale
  resp=
  | POST https://api.eu-west-2.outscale.com/api/v1/ReadPublicIps
  | Status: 401 (245.207039ms)
  |
  | {
  |   "Errors": [
  |     {
  |       "Code": "1",
  |       "Details": "",
  |       "Type": "AccessDenied"
  |     }
  |   ],
  |   "ResponseContext": {
  |     "RequestId": "34202855-6cda-41f8-946f-f712bf6e7e54"
  |   }
  | }
```

### Crash Output (if applicable)

_No response_

### Expected Behavior

200

### Actual Behavior

401

### Steps to Reproduce

1. configure outscale provider 1.6.0 with env vars
2. add a random simple configuration like
```
resource "outscale_public_ip" "public_ip01" {
}
```
3. tofu plan / apply

### Additional Context

Works great in 1.5.0. Read the changelog and didn't see anything related to environment variables. Provider doc still present this way of authenticating https://registry.terraform.io/providers/outscale/outscale/latest/docs#environment-variables.

Regards,

### References

_No response_

--- a/MIGRATION.md
+++ b/MIGRATION.md
@@ -24,6 +24,18 @@ Before starting, **always back up your Terraform state and configuration files.*
   In versions **< v0.9.0**, creating an `access_key` **without an expiration date** is no longer supported.
   See [Issue #342](https://github.com/outscale/terraform-provider-outscale/issues/342) for more details.
 
+- **Provider environment variables:**
+  Starting from version **v1.6.0**, the legacy `OUTSCALE_*` environment variables have been replaced by `OSC_*` variables.
+
+| Legacy variable supported up to `v1.5.0` | Variable to use in `v1.6.0+` | Provider configuration |
+| --- | --- | --- |
+| `OUTSCALE_ACCESSKEYID` | `OSC_ACCESS_KEY` | `access_key_id` |
+| `OUTSCALE_SECRETKEYID` | `OSC_SECRET_KEY` | `secret_key_id` |
+| `OUTSCALE_REGION` | `OSC_REGION` | `api.region`, `oks.region` |
+| `OUTSCALE_X509CERT` | `OSC_X509_CLIENT_CERT` | `api.x509_cert_path` |
+| `OUTSCALE_X509KEY` | `OSC_X509_CLIENT_KEY` | `api.x509_key_path` |
+| `OUTSCALE_OAPI_URL` | `OSC_ENDPOINT_API` | `api.endpoint` |
+
 ---
 
 ## 📝 Migration Steps

--- a/README.md
+++ b/README.md
@@ -94,6 +94,8 @@ terraform refresh
 There is a breaking change when creating an `access_key` without expiration date in versions `< v0.9.0`.
 See [Issue #342](https://github.com/outscale/terraform-provider-outscale/issues/342).
 
+Starting from version `v1.6.0`, the legacy `OUTSCALE_*` environment variables have been replaced by `OSC_*` variables documented in the [provider registry documentation](./docs/index.md#environment-variables).
+
 ---
 
 ## 🚀 Using the Provider
@@ -213,8 +215,8 @@ provider "outscale" {
 
 Or set environment variables:
 ```sh
-export OUTSCALE_X509CERT=/myrepository/certificate/client_ca.crt
-export OUTSCALE_X509KEY=/myrepository/certificate/client_ca.key
+export OSC_X509_CLIENT_CERT=/myrepository/certificate/client_ca.crt
+export OSC_X509_CLIENT_KEY=/myrepository/certificate/client_ca.key
 ```
 
 ---

--- a/docs/index.md
+++ b/docs/index.md
@@ -65,7 +65,9 @@ provider "outscale" {
 
 ### Environment variables
 
-You can provide your credentials with the `OUTSCALE_ACCESSKEYID` and `OUTSCALE_SECRETKEYID` environment variables:
+You can provide your credentials with the `OSC_ACCESS_KEY` and `OSC_SECRET_KEY` environment variables.
+
+Starting from version `v1.6.0`, the legacy `OUTSCALE_*` environment variables have been replaced by `OSC_*` variables.
 
 Example:
 
@@ -76,14 +78,25 @@ provider "outscale" {}
 Usage:
 
 ```console
-$ export OUTSCALE_ACCESSKEYID="myaccesskey"
-$ export OUTSCALE_SECRETKEYID="mysecretkey"
-$ export OUTSCALE_REGION="cloudgouv-eu-west-1"
-$ export OUTSCALE_X509CERT="~/certificate/certificate.crt"
-$ export OUTSCALE_X509KEY="~/certificate/certificate.key"
+$ export OSC_ACCESS_KEY="myaccesskey"
+$ export OSC_SECRET_KEY="mysecretkey"
+$ export OSC_REGION="cloudgouv-eu-west-1"
+$ export OSC_X509_CLIENT_CERT="~/certificate/certificate.crt"
+$ export OSC_X509_CLIENT_KEY="~/certificate/certificate.key"
 
 $ terraform plan
 ```
+
+Migration from the legacy provider environment variables to the SDK environment variables:
+
+| Legacy variable supported up to `v1.5.0` | Variable to use in `v1.6.0+` | Provider configuration |
+| --- | --- | --- |
+| `OUTSCALE_ACCESSKEYID` | `OSC_ACCESS_KEY` | `access_key_id` |
+| `OUTSCALE_SECRETKEYID` | `OSC_SECRET_KEY` | `secret_key_id` |
+| `OUTSCALE_REGION` | `OSC_REGION` | `api.region`, `oks.region` |
+| `OUTSCALE_X509CERT` | `OSC_X509_CLIENT_CERT` | `api.x509_cert_path` |
+| `OUTSCALE_X509KEY` | `OSC_X509_CLIENT_KEY` | `api.x509_key_path` |
+| `OUTSCALE_OAPI_URL` | `OSC_ENDPOINT_API` | `api.endpoint` |
 
 ## Configuration
 
@@ -124,23 +137,23 @@ In addition to [generic provider arguments](https://www.terraform.io/docs/config
 
 * `config_file` - (Optional) The path to an OSC config file. It can also be sourced from the `OSC_CONFIG_FILE` [environment variable](#set-a-profile-using-environment-variables).
 * `profile` - (Optional) The named profile you want to use in the OSC config file. It can also be sourced from the `OSC_PROFILE` [environment variable](#set-a-profile-using-environment-variables).
-* `access_key_id` - (Optional) The ID of the OUTSCALE access key. It must be provided, but it can also be sourced from the `OUTSCALE_ACCESSKEYID` [environment variable](#environment-variables).
-* `secret_key_id` - (Optional) The OUTSCALE secret key. It must be provided, but it can also be sourced from the `OUTSCALE_SECRETKEYID` [environment variable](#environment-variables).
+* `access_key_id` - (Optional) The ID of the OUTSCALE access key. It must be provided, but it can also be sourced from the `OSC_ACCESS_KEY` [environment variable](#environment-variables).
+* `secret_key_id` - (Optional) The OUTSCALE secret key. It must be provided, but it can also be sourced from the `OSC_SECRET_KEY` [environment variable](#environment-variables).
 * `api` - (Optional) Configuration elements for OUTSCALE API operations.
-    * `endpoint` - (Optional) The endpoint to use for OUTSCALE API operations. For more information on available endpoints, see [API Endpoints Reference > OUTSCALE API](https://docs.outscale.com/en/userguide/API-Endpoints-Reference.html#_outscale_api).
-    * `region` - (Optional) The Region to use for OUTSCALE API operations. It can also be sourced from the `OUTSCALE_REGION` [environment variable](#environment-variables). For more information on available Regions, see [About Regions and Subregions](https://docs.outscale.com/en/userguide/About-Regions-and-Subregions.html).
-    * `x509_cert_path` - (Optional) The path to the x509 Client Certificate. It can also be sourced from the `OUTSCALE_X509CERT` [environment variable](#environment-variables). For more information on the use of those certificates, see [About API Access Rules](https://docs.outscale.com/en/userguide/About-API-Access-Rules.html).
-    * `x509_key_path` - (Optional) The path to the private key of the x509 Client Certificate. It can also be sourced from the `OUTSCALE_X509KEY` [environment variable](#environment-variables). For more information on the use of those certificates, see [About API Access Rules](https://docs.outscale.com/en/userguide/About-API-Access-Rules.html).
+    * `endpoint` - (Optional) The endpoint to use for OUTSCALE API operations. It can also be sourced from the `OSC_ENDPOINT_API` [environment variable](#environment-variables). For more information on available endpoints, see [API Endpoints Reference > OUTSCALE API](https://docs.outscale.com/en/userguide/API-Endpoints-Reference.html#_outscale_api).
+    * `region` - (Optional) The Region to use for OUTSCALE API operations. It can also be sourced from the `OSC_REGION` [environment variable](#environment-variables). For more information on available Regions, see [About Regions and Subregions](https://docs.outscale.com/en/userguide/About-Regions-and-Subregions.html).
+    * `x509_cert_path` - (Optional) The path to the x509 Client Certificate. It can also be sourced from the `OSC_X509_CLIENT_CERT` [environment variable](#environment-variables). For more information on the use of those certificates, see [About API Access Rules](https://docs.outscale.com/en/userguide/About-API-Access-Rules.html).
+    * `x509_key_path` - (Optional) The path to the private key of the x509 Client Certificate. It can also be sourced from the `OSC_X509_CLIENT_KEY` [environment variable](#environment-variables). For more information on the use of those certificates, see [About API Access Rules](https://docs.outscale.com/en/userguide/About-API-Access-Rules.html).
     * `insecure` - (Optional) Enables TLS insecure connection.
 * `oks` - (Optional) Configuration elements for OKS API operations.
-    * `endpoint` - (Optional) The endpoint to use for OKS API operations. For more information on available endpoints, see [API Endpoints Reference > OUTSCALE Kubernetes as a Service (OKS)](https://docs.outscale.com/en/userguide/API-Endpoints-Reference.html#_outscale_kubernetes_as_a_service_oks).
-    * `region` - (Optional) The Region to use for OKS API operations. It can also be sourced from the `OUTSCALE_REGION` [environment variable](#environment-variables). For more information on available Regions, see [About Regions and Subregions](https://docs.outscale.com/en/userguide/About-Regions-and-Subregions.html).
+    * `endpoint` - (Optional) The endpoint to use for OKS API operations. It can also be sourced from the `OSC_ENDPOINT_OKS` [environment variable](#environment-variables). For more information on available endpoints, see [API Endpoints Reference > OUTSCALE Kubernetes as a Service (OKS)](https://docs.outscale.com/en/userguide/API-Endpoints-Reference.html#_outscale_kubernetes_as_a_service_oks).
+    * `region` - (Optional) The Region to use for OKS API operations. It can also be sourced from the `OSC_REGION` [environment variable](#environment-variables). For more information on available Regions, see [About Regions and Subregions](https://docs.outscale.com/en/userguide/About-Regions-and-Subregions.html).
 
 The following top-level arguments are deprecated but still supported as a fallback:
 * `endpoints` - (Optional, deprecated) The endpoints to use for OUTSCALE API and OKS API operations. For more information on available endpoints, see [Regions, Endpoints and Availability Zones Reference](https://docs.outscale.com/en/userguide/Regions-Endpoints-and-Availability-Zones-Reference.html).
     * `api` - (Optional, deprecated) For OUTSCALE API.
     * `oks` - (Optional, deprecated) For OKS API.
-* `region` - (Optional, deprecated) The Region to use for OUTSCALE API and OKS API operations. It can also be sourced from the `OUTSCALE_REGION` [environment variable](#environment-variables). For more information on available Regions, see [About Regions and Subregions](https://docs.outscale.com/en/userguide/About-Regions-and-Subregions.html).
-* `x509_cert_path` - (Optional, deprecated) The path to the x509 Client Certificate. It can also be sourced from the `OUTSCALE_X509CERT` [environment variable](#environment-variables). For more information on the use of those certificates, see [About API Access Rules](https://docs.outscale.com/en/userguide/About-API-Access-Rules.html).
-* `x509_key_path` - (Optional, deprecated) The path to the private key of the x509 Client Certificate. It can also be sourced from the `OUTSCALE_X509KEY` [environment variable](#environment-variables). For more information on the use of those certificates, see [About API Access Rules](https://docs.outscale.com/en/userguide/About-API-Access-Rules.html).
+* `region` - (Optional, deprecated) The Region to use for OUTSCALE API and OKS API operations. It can also be sourced from the `OSC_REGION` [environment variable](#environment-variables). For more information on available Regions, see [About Regions and Subregions](https://docs.outscale.com/en/userguide/About-Regions-and-Subregions.html).
+* `x509_cert_path` - (Optional, deprecated) The path to the x509 Client Certificate. It can also be sourced from the `OSC_X509_CLIENT_CERT` [environment variable](#environment-variables). For more information on the use of those certificates, see [About API Access Rules](https://docs.outscale.com/en/userguide/About-API-Access-Rules.html).
+* `x509_key_path` - (Optional, deprecated) The path to the private key of the x509 Client Certificate. It can also be sourced from the `OSC_X509_CLIENT_KEY` [environment variable](#environment-variables). For more information on the use of those certificates, see [About API Access Rules](https://docs.outscale.com/en/userguide/About-API-Access-Rules.html).
 * `insecure` - (Optional, deprecated) Enables TLS insecure connection.

